### PR TITLE
go-server: Support additional properties as command line arguments

### PR DESCRIPTION
**Pull request by mapero**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [x] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `3.4.x`, `4.0.x`. Default: `master`.
- [x] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR

With this Pull Requests additional properties are supported as command line arguments:

- packageVersion
- sourceFolder
- serverPort

Thanks.

@antihax (2017/11)
@bvwells (2017/12)
@grokify (2018/07)
@kemokemo (2018/09)

### CODE REVIEW

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/GoServerCodegen.java [hunk 1]
```diff
@@ -31,16 +32,26 @@
 
     private static final Logger LOGGER = LoggerFactory.getLogger(GoServerCodegen.class);
 
-    protected String apiVersion = "1.0.0";
-    protected int serverPort = 8080;
+    protected String packageVersion = "1.0.0";
+    protected String serverPort = "8080";
```
Comment (wing328) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/GoServerCodegen.java]: @mapero Thanks for the contribution. should the port number be an integer instead?

**Comment by wing328** [comment]
Thanks for the PR but your commit (as shown in the Commits tab) is not linked to your Github account, which means this PR won't count as your contribution in https://github.com/OpenAPITools/openapi-generator/graphs/contributors.

Let me know if you need help fixing it.

Ref: https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-can-i-update-commits-that-are-not-linked-to-my-github-account

**Comment by bvwells** [comment]
Hi @mapero, thanks for the PR. 

The Go client generator already has the PACKAGE_VERSION as a command line option so I'd recommend that you move it to the Go abstract base class and remove from the client generator. See [here](https://github.com/OpenAPITools/openapi-generator/blob/e3ab0595307884bc17bce602b8ac9bf294a29260/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractGoCodegen.java#L113) and [here](https://github.com/OpenAPITools/openapi-generator/blob/e3ab0595307884bc17bce602b8ac9bf294a29260/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/GoClientCodegen.java#L56).

Let me know if you need any help with this.

**Comment by mapero** [comment]
> 
> 
> Hi @mapero, thanks for the PR.
> 
> The Go client generator already has the PACKAGE_VERSION as a command line option so I'd recommend that you move it to the Go abstract base class and remove from the client generator. See [here](https://github.com/OpenAPITools/openapi-generator/blob/e3ab0595307884bc17bce602b8ac9bf294a29260/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractGoCodegen.java#L113) and [here](https://github.com/OpenAPITools/openapi-generator/blob/e3ab0595307884bc17bce602b8ac9bf294a29260/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/GoClientCodegen.java#L56).
> 
> Let me know if you need any help with this.

I have updated Client and Server to share the cli option for PACKAGE_VERSION

**Comment by mapero** [comment]
@bvwells Done
@wing328 Done

Should be ready to merge.

**Comment by ackintosh** [comment]
I've updated the changes to resolve conflicts. waiting for the CI. 👷

**Comment by wing328** [comment]
The CircleCI issue seems to be transient. Let's see if it occurs again after merging the PR into master

**Pull request merged**